pkg/restapi/issuer/operation/operations.go

```diff
@@ -1139,22 +1148,77 @@ func (o *Operation) storeUserInvitationMapping(mapping *UserInvitationMapping) e
 }
 
 // TODO fulfillment shouldn't be saved, need re-design [Issue#560]
-func (o *Operation) saveCredentialFulfillment(thID string, fulfillment *verifiable.Presentation) error {
-	fbytes, err := fulfillment.MarshalJSON()
+func (o *Operation) saveCredentialAttachments(thID string, attachments *credentialAttachments) error {
+	manifestBytes, err := json.Marshal(attachments.Manifest)
 	if err != nil {
-		return fmt.Errorf("failed to marshal credential fulfillment: %w", err)
+		return fmt.Errorf("failed to marshal credential manifest: %w", err)
 	}
 
-	err = o.txnStore.Put(thID, fbytes)
+	fulfilmentBytes, err := json.Marshal(attachments.Fulfilment)
+	if err != nil {
+		return fmt.Errorf("failed to marshal credential fulfilment: %w", err)
+	}
+
+	err = o.txnStore.Batch([]storage.Operation{
+		{
+			Key:   manifestPrefix + thID,
+			Value: manifestBytes,
+		},
+		{
+			Key:   fulfilmentPrefix + thID,
+			Value: fulfilmentBytes,
+		},
+	})
 	if err != nil {
-		return fmt.Errorf("failed to save credential fulfillment in store: %w", err)
+		return fmt.Errorf("failed to save credential attachments in store: %w", err)
 	}
 
 	return nil
 }
 
+func (o *Operation) readAndValidateCredentialApplication(msg service.DIDCommAction) (*cm.CredentialApplication, error) {
+	applicationAttachment, err := getAttachments(msg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to correlate WACI interaction, missing thread ID: %w", err)
```

Comment:
Incorrect error message - should be "failed to get attachments" or something similar